tensorcircuit/mpscircuit.py

```diff
@@ -79,71 +88,120 @@ class MPSCircuit(AbstractCircuit):
 
     """
 
-    # TODO(@SUSYUSTC): Add 3-qubits gates support
+    # TODO(@SUSYUSTC): fix the jax backend performance issue
 
     is_mps = True
 
     def __init__(
         self,
         nqubits: int,
+        center_position: Optional[int] = None,
         tensors: Optional[Sequence[Tensor]] = None,
-        center_position: int = 0,
+        wavefunction: Optional[Union[QuVector, Tensor]] = None,
+        **rules: Any,
     ) -> None:
         """
         MPSCircuit object based on state simulator.
 
         :param nqubits: The number of qubits in the circuit.
         :type nqubits: int
-        :param tensors: If not None, the initial state of the circuit is taken as ``tensors``
-            instead of :math:`\\vert 0\\rangle^n` qubits, defaults to None
-        :type tensors: Sequence[Tensor], optional
         :param center_position: The center position of MPS, default to 0
         :type center_position: int, optional
-        """
-        if tensors is None:
+        :param tensors: If not None, the initial state of the circuit is taken as ``tensors``
+            instead of :math:`\\vert 0\\rangle^n` qubits, defaults to None.
+        When ``tensors`` are specified, if ``center_position`` is None, then the tensors are canonicalized,
+            otherwise it is assumed the tensors are already canonicalized at the ``center_position``
+        :type tensors: Sequence[Tensor], optional
+        :param wavefunction: If not None, it is transformed to the MPS form according to the truncation rules
+        :type wavefunction: Tensor
+        :param rules: Truncation rules
+        :type rules: Any
+        """
+        self.circuit_param = {
+            "nqubits": nqubits,
+            "center_position": center_position,
+            "rules": rules,
+            "tensors": tensors,
+            "wavefunction": wavefunction,
+        }
+        self.rules = rules
+        if wavefunction is not None:
+            assert (
+                tensors is None
+            ), "tensors and wavefunction cannot be used at input simutaneously"
+            if isinstance(wavefunction, QuVector):
+                wavefunction = wavefunction.eval()
```

Comment:
should be translated to FiniteMPS directly without eval: exponential memory to eval

Reply:
pls add a TODO for remind

Reply:
done